js/layout-javascript/agenda-code.js

```diff
@@ -590,23 +655,32 @@ DynamicList.prototype.attachObservers = function() {
         return;
       }
     })
-    .on('click', '.agenda-date-selector li', function() {
-      // prevents clicking the active one
-      // prevents clicking the placeholder
-      if ($(this).hasClass('active') || $(this).hasClass('placeholder')) {
+    .on('keydown', '.agenda-date-selector li', function(event) {
+      if (!_this.Utils.accessibilityHelpers.isExecute(event) || $(this).is('.active, .placeholder')) {
         return;
       }
 
-      var indexOfActiveDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(_this.$container.find('.agenda-date-selector li.active'));
-      var indexOfClickedDate = _this.$container.find('.agenda-date-selector li').not('.placeholder').index(this);
+      var indexOfActiveDate = _this.$container
+        .find('.agenda-date-selector li')
+        .not('.placeholder')
+        .index(_this.$container.find('.agenda-date-selector li.active'));
+      var indexOfClickedDate = _this.$container
+        .find('.agenda-date-selector li')
+        .not('.placeholder')
+        .index(this);
       var indexDifference = indexOfClickedDate - indexOfActiveDate;
 
       _this.updateDateIndexContext(indexOfClickedDate);
 
       Fliplet.Analytics.trackEvent({
         category: 'list_dynamic_' + _this.data.layout,
         action: 'filter_date',
-        label: $(this).find('.week').text().trim() + ' ' + $(this).find('.day').text().trim() + ' ' + $(this).find('.month').text().trim()
+        label:
```

Comment:
Let's use an array joined by space, and use multiple lines of code if necessary.

```js
[$(this).find('.week').text().trim(), $(this).find('.day').text().trim(), $(this).find('.month').text().trim()].join(' ')
```

Reply:
@tonytlwu fixed